packages/hash/realtime/src/index.ts

```diff
@@ -45,23 +43,48 @@ const logger = new Logger({
 
 const QUEUES = generateQueues(logger);
 
-const acquireSlot = async (pool: PgPool) => {
+const shutdown = new GracefulShutdown(logger, "SIGINT", "SIGTERM");
+
+const acquireReplicationSlot = async (
+  pool: PgPool,
+  slotName: string,
+  options: { temporary?: boolean },
+) => {
   // Create the slot if it does not exist
+  await pool.query(sql`
```

Comment:
This moves the "get or create" semantics to the database instead of typescript, inspired by https://github.com/supabase/realtime/blob/5dcf96b3629aa98ba55cc212aea8bb1fbeedf4af/server/lib/realtime/rls/replications.ex#L4-L18

Their product is really good for the same purposes we have realtime, so borrowed some of their setup as it's proven in production. This includes making slots "temporary" instead of permanent to prevent disk space creep and orphaned replication slots

Reply:
Do we perhaps want to link to this as inspiration in the code

Reply:
Of course we should, added in ef6c339c9a9af27528bd904250029835fe59137f